### PR TITLE
Add href to nav and header links

### PR DESCRIPTION
**Pull request by 0618**
#### Description of changes

**Issue**: Our links are missing `href`.
CleanShot 2022-02-02 at 13 48 57@2x

**Code Change**:
- add `href`
- refactor to use `React.forwardRef` as suggested in [Nextjs's doc](https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-functional-component)
- add missing `passHref` prop as suggested in [Nextjs's doc](https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag)
- fix `NevLink` type error
Screen Shot 2022-03-02 at 13 30 47
- refactor `import` in `Header.tsx` and `LinkButton.tsx` for tree-shaking

#### Issue #, if available

https://app.asana.com/0/1201736086077838/1201759796183693/f

#### Description of how you validated changes

- ran `yarn test`
- local test

image

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/components/Layout/LinkButton.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    children?: ReactElement;
+  },
+  ref
+) => (
+  <a onClick={onClick} className={classNames} href={href} ref={ref}>
+    {children}
+  </a>
+);
+
+export default forwardRef(LinkButton);
```
Comment (dbanksdesign) [inline docs/src/components/Layout/LinkButton.tsx:23]: Nice to break this out to a separate file!

File: docs/src/components/Layout/Header.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import { useState } from 'react';
```
Comment (dbanksdesign) [inline docs/src/components/Layout/Header.tsx:1]: I want to say there is a reason we use `import * as React from 'react';` as opposed to `import {} from 'react';` but I don't remember why.

**Review by dbanksdesign — APPROVED** [review]
LGTM! One general question for the team about importing react, but otherwise lets get this in!

File: docs/src/components/Layout/Header.tsx (hunk 2 above)
Comment (reesscot) [reply docs/src/components/Layout/Header.tsx:1]: We prefer the `import * as React from 'react';` for consistency across the codebase and minimizing import statement code changes. See: 
https://epicreact.dev/importing-react-through-the-ages/

File: docs/src/components/Layout/Header.tsx [hunk 3]
```diff
@@ -21,8 +21,20 @@ import { useRouter } from 'next/router';
 import { Logo } from '@/components/Logo';
 import { FrameworkChooser } from './FrameworkChooser';
 import { SecondaryNav } from './SecondaryNav';
+import LinkButton from './LinkButton';
+import type { MouseEventHandler, ReactElement } from 'react';
```
Comment (reesscot) [inline docs/src/components/Layout/Header.tsx:25]: Nice, love the `import type`

File: docs/src/components/Layout/Header.tsx [hunk 4]
```diff
@@ -21,8 +21,20 @@ import { useRouter } from 'next/router';
...
+const NavLink = ({
+  href,
+  children,
+  isExternal = false,
+  onClick,
+}: {
+  href: string;
+  children: ReactElement;
+  isExternal?: boolean;
+  onClick?: MouseEventHandler<HTMLAnchorElement>;
```
Comment (reesscot) [inline docs/src/components/Layout/Header.tsx:36]: Thanks for fixing the types

**Review by reesscot — APPROVED** [review]
LGTM. Would prefer to keep the react namespace import as it was, but not a blocker.

File: docs/src/components/Layout/Header.tsx (hunk 2 above)
Comment (0618) [reply docs/src/components/Layout/Header.tsx:1]: Thanks for sharing! It's fun to know all the histories on how to import react.

Looks like both are fine, but I'll convert my change back for consistency :D

File: docs/src/components/Layout/Header.tsx (hunk 3 above)
Comment (wlee221) [reply docs/src/components/Layout/Header.tsx]: I love this too! Nice JS/TS distinguishment.

**Review by wlee221 — APPROVED** [review]

**Pull request merged**